Desktop/widgets/factorlevellistbase.h

```diff
@@ -16,64 +16,77 @@
 public:
-	RepeatedMeasuresFactorsListBase(QQuickItem* parent = nullptr);
+	FactorLevelListBase(QQuickItem* parent = nullptr);
 
 	bool			isJsonValid(const Json::Value& optionValue)	override;
 	Json::Value		createJson()								override;
 	void			bindTo(const Json::Value& value)			override;
 	bool			encodeValue()						const	override	{ return true; }
-	ListModel*		model()								const	override	{ return _factorsModel; }
+	ListModel*		model()								const	override	{ return _factorLevelsModel; }
 	void			setUpModel()								override;
 	void			setUp()										override;
 
-	QStringList		defaultFactors()					const				{ return _defaultFactors;		}
-	QStringList		defaultLevels()						const				{ return _defaultLevels;		}
+	QString			factorName()						const				{ return _factorName;			}
+	QString			levelName()							const				{ return _levelName;			}
 	QString			factorPlaceHolder()					const				{ return _factorPlaceHolder;	}
 	QString			levelPlaceHolder()					const				{ return _levelPlaceHolder;		}
+	int				minFactors()						const				{ return _minFactors;			}
+	int				minLevels()							const				{ return _minLevels;			}
+
+	QString			getFactorName(int i)				const				{ return QStringLiteral("%1 %2").arg(_factorName).arg(i);	}
+	QString			getLevelName(int i)					const				{ return QStringLiteral("%1 %2").arg(_levelName).arg(i);	}
 
 signals:
 	void			itemChanged(int index, QString name);
 	void			itemRemoved(int index);
 
-	void			defaultFactorsChanged();
-	void			defaultLevelsChanged();
+	void			factorNameChanged();
+	void			levelNameChanged();
 	void			factorPlaceHolderChanged();
 	void			levelPlaceHolderChanged();
+	void			minFactorsChanged();
+	void			minLevelsChanged();
 
 protected slots:
 	void			termsChangedHandler() override;
 
 protected:
-	GENERIC_SET_FUNCTION(DefaultFactors,		_defaultFactors,		defaultFactorsChanged,		QStringList		)
-	GENERIC_SET_FUNCTION(DefaultLevels,			_defaultLevels,			defaultLevelsChanged,		QStringList		)
-	GENERIC_SET_FUNCTION(FactorPlaceHolder,		_factorPlaceHolder,		factorPlaceHolderChanged,	QString			)
-	GENERIC_SET_FUNCTION(LevelPlaceHolder,		_levelPlaceHolder,		levelPlaceHolderChanged,	QString			)
+	GENERIC_SET_FUNCTION(FactorName,			_factorName,			factorNameChanged,			QString		)
+	GENERIC_SET_FUNCTION(LevelName,				_levelName,				levelNameChanged,			QString		)
+	GENERIC_SET_FUNCTION(FactorPlaceHolder,		_factorPlaceHolder,		factorPlaceHolderChanged,	QString		)
+	GENERIC_SET_FUNCTION(LevelPlaceHolder,		_levelPlaceHolder,		levelPlaceHolderChanged,	QString		)
+	GENERIC_SET_FUNCTION(MinFactors,			_minFactors,			minFactorsChanged,			int			)
+	GENERIC_SET_FUNCTION(MinLevels,				_minLevels,				minLevelsChanged,			int			)
 
 private:
-	ListModelRepeatedMeasuresFactors*	_factorsModel	= nullptr;
+	ListModelFactorLevels*				_factorLevelsModel	= nullptr;
 
-	QStringList							_defaultFactors		=	{ tr("RM Factor 1")					};
-	QStringList							_defaultLevels		=	{ tr("Level 1"), tr("Level 2")	};
+	QString								_factorName			=	tr("Factor");
```

Comment:
This will only be set in the constructor, no?

So when the language changes this isn't in fact updated is it?
Or does that happen because QML is reloaded completely and thus the constructor as well?

Reply:
It is even worse: tr() does not work in a header file.
But it is indeed better to set the default value is QML: it will be changed automatically when the language is changed. This won't be the case in the constructor (even in the c++ file).